Converted log collection form to Angular

AparnaKarve opened the pull request:
The log collection form is the 3rd form that deals with FileDepot and now undergoing AngularJS conversion.
The other two forms Schedule Editor and Diagnostics Database are already converted to AngularJS.

https://bugzilla.redhat.com/show_bug.cgi?id=1217348

AparnaKarve commented:
@brandondunne This PR should fix the issue that you were seeing with log collection.

brandondunne commented on vmdb/app/controllers/ops_controller/diagnostics.rb:
```diff
@@ -1164,4 +1160,9 @@ def diagnostics_build_tree
     TreeBuilderOpsDiagnostics.new("diagnostics_tree", "diagnostics", @sb)
   end
 
+  def build_supported_depots_for_select
```
Sorry, many comments...
- Is @protocols_arr used anywhere else?
- Each with push could just be collect
- Overall it looks like this could just be ```@supported_depots_for_select = FileDepot.supported_depots.values.sort```

AparnaKarve replied:
@brandondunne Made your suggested change -
```
@supported_depots_for_select = FileDepot.supported_depots.values.sort
```
Thanks.

chessbyte commented on vmdb/app/controllers/ops_controller.rb:
```diff
@@ -680,7 +681,7 @@ def replace_right_cell(nodetype, replace_trees = false) # replace_trees can be a
             page.replace("ops_tabs", :partial=>"all_tabs")
           elsif nodetype == "log_depot_edit"
             @right_cell_text = "Editing Log Depot settings"
-            page.replace_html("diagnostics_collect_logs", :partial=>"layouts/edit_log_depot_settings")
```
Should the partial `layouts/edit_log_depot_settings` be deleted as well or is it still being used?

AparnaKarve replied:
@chessbyte I verified that the log collection was the last piece using this partial, which means it can now be removed in this PR.
In the last update that I pushed, I have removed this partial. Thanks for the suggestion.

jrafanie replied:
:+1: :red_circle: :red_circle: :red_circle:

brandondunne commented on vmdb/spec/models/miq_schedule_spec.rb:
```diff
@@ -684,7 +684,12 @@
   end
 
   context "#verify_file_depot" do
-    let(:params) { {:uri_prefix => "ftp", :uri => "ftp://ftp.example.com", :name => "Test Backup Depot", :username => "user", :password => "password"} }
```
Why was this taken out of the "let"?

AparnaKarve replied:
@brandondunne If you recall, the test was failing earlier only on Travis and was working fine locally, thats when I had made a few changes in the test ... the main goal was to identify the root cause of why it was failing only on Travis.
I have now restored all those changes, since we have identified the root cause.

Fryguy commented on vmdb/app/assets/javascripts/services/miq_db_backup_service.js:
```diff
@@ -26,7 +28,14 @@ miqAngularApplication.service('miqDBBackupService', function() {
+        if(model.log_protocol == 'Samba' || model.log_protocol == 'FTP')
+            return true;
+        else
+            return false;
```
This indent seems really big...what's the style standard here?

AparnaKarve replied:
@Fryguy Fixed the indent.

kbrock commented on vmdb/app/assets/javascripts/services/miq_db_backup_service.js:
```diff
@@ -26,7 +28,14 @@ miqAngularApplication.service('miqDBBackupService', function() {
+  };
+
+  this.credsProtocol = function(model) {
+    if(model.log_protocol == 'Samba' || model.log_protocol == 'FTP')
```
```ruby
%w(Samba FTP).include?(model.log_protocol)
```

but only change if you are making other changes, too.

brandondunne commented:
@AparnaKarve One concern:  Selecting a "RedHatDropbox" depot should not prompt for depot_name or uri.

Everything else appears to be working.

kbrock commented on vmdb/app/controllers/ops_controller.rb:
```diff
@@ -680,7 +681,7 @@ def replace_right_cell(nodetype, replace_trees = false) # replace_trees can be a
           elsif nodetype == "log_depot_edit"
             @right_cell_text = "Editing Log Depot settings"
-            page.replace_html("diagnostics_collect_logs", :partial=>"layouts/edit_log_depot_settings")
+            page.replace_html("diagnostics_collect_logs", :partial => "ops/log_collection")
```
is this rjs?

AparnaKarve replied:
This is just a regular haml with AngularJS hooks in it.

AparnaKarve commented:
@brandondunne Ensured that Depot Name and Uri are hidden for "Red Hat Dropbox".
Please verify. Thanks.

brandondunne commented:
The UI also used to show a text box for "support case" on Red Hat Dropbox, sorry I forgot about that.

Also, the FileDepot that was created is of the right type, but it has the wrong values.  Are you sending blank strings?
```
=> #<FileDepotFtpAnonymousRedhatDropbox id: 165, name: "", uri: "ftp://", created_at: "2015-05-04 19:52:12", updated_at: "2015-05-04 19:52:12", type: "FileDepotFtpAnonymousRedhatDropbox", support_case: nil>
```

Are there any tests to go along with this?  Remembering all of these things and ensuring everything is tested by clicking on it is very error prone. :worried:

jrafanie commented:
Good point @brandondunne, we definitely need tests for the new converted areas to ensure they remain testable going forward.

AparnaKarve commented:
@brandondunne I verified that the UI still shows the text box for "support case" on Red Hat Dropbox.
Please confirm if you see it.
I also fixed the JS that was sending blank param values.
Since the testing around this is so UI-centric, I have added new Jasmine specs for a few functions to check if only the necessary params have values in it and the rest are null so that they do not end up at the backend.

brandondunne commented:
Things appear to be ok clicking around in the UI.
Tested:
- Entering edit screen for a zone without log collection configured
- Setting log collection type for Red Hat Dropbox
- Entering edit screen for a zone configured for log collection
- Changing to a different log collection type.
- Validate button calls the correct model to validate credentials
- Saving changing depot type
- Canceling edit
- Available fields change based on the type

jrafanie commented:
@dclarizio @AparnaKarve do we have tests for the several hundred lines of javascript added in this PR?

AparnaKarve commented:
@jrafanie Yes, there are Jasmine tests added for the JS code
Check out - 
```
log_collection_form_controller_spec.js
miq_db_backup_service_spec.js
miq_service_spec.js
```

jrafanie commented:
Great, thanks @AparnaKarve, I guess I'm blind... not sure why I didn't see them. :+1:

AparnaKarve commented:
No worries @jrafanie , it happens to the best of us.
